### PR TITLE
[v1.x] Fix nightly CD for python docker image releases

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
This PR makes multiple changes to fix the nightly CD pipeline for python docker image releases.
- Cherry-pick https://github.com/apache/incubator-mxnet/commit/34010ea417014a6b6203395d3359099431c362cf
- Change push target from dockerhub to ecr public repository
- Update CUDA 11.2 base image for ubuntu16
- Use nightly_v1.x prefix to differentiate from master branch docker tags
- Remove unused mxnet_lib/dynamic pipeline

### CODE REVIEW

**Review by szha — APPROVED** [review]

**Review by leezu — COMMENTED** [review]
As part of this work, can we reconsider the publication strategy? Currently nightly builds are pushed to dockerhub, which is not compatible with ASF policy. Instead, can you configure the job to only push on approved release tags?

**Comment by szha** [comment]
we could also consider publishing nightly tags to ECR and make them available to only developers.

**Comment by mseth10** [comment]
@leezu @szha We have changed docker push target to a public ecr repo in master with this PR https://github.com/apache/incubator-mxnet/pull/19772  . Will backport it to v1.x too.

**Review by access2rohit — APPROVED** [review]
LGTM

File: cd/Jenkinsfile_utils.groovy [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
-// 'Jenkins_pipeline.groovy' file and has the pipeline definition for the 
+// The 'job_type' should be the name of the directory that contains the
+// 'Jenkins_pipeline.groovy' file and has the pipeline definition for the
 // artifact (docker image, binary, pypi or maven package, etc.) that should
 // be published.
 
 STATE_UPDATE="State Update"
 
-def trigger_release_job(job_name, job_type, mxnet_variants) {
+def trigger_release_job(cd_release_job, job_name, job_type, mxnet_variants) {
```
Comment (waytrue17) [inline cd/Jenkinsfile_utils.groovy:29]: Curious, why do we pass the cd_release_job instead of using the env var?

**Review by waytrue17 — APPROVED** [review]

File: cd/Jenkinsfile_utils.groovy (hunk 1 above)
Comment (mseth10) [reply cd/Jenkinsfile_utils.groovy:29]: CD_RELEASE_JOB_NAME is actually a parameter of the CD pipeline, so this is how it's supposed to be used. It is also set as a global env variable, which can potentially cause some issues.
For reference, it is addressed in master by this PR https://github.com/apache/incubator-mxnet/pull/17775

**Pull request merged**